system/lib/libc/musl/src/unistd/dup2.c

```diff
@@ -13,10 +13,10 @@ int dup2(int old, int new)
 	while ((r=__syscall(SYS_dup2, old, new))==-EBUSY);
 #else
 	if (old==new) {
-#if __EMSCRIPTEN__
-		r = __syscall(SYS_fcntl, old, F_GETFD);
+#ifdef __EMSCRIPTEN__
+		r = __wasi_fd_is_valid(old) ? 0 : -1;
 #else
-		r = __wasi_fd_is_valid(old);
+		r = __syscall(SYS_fcntl, old, F_GETFD);
```

Comment:
We should probably just remove this patch completely then.. if SYS_dup2 is always defined under emscripten